docs/src/user-guide/reference-json-search-syntax.md

```diff
@@ -0,0 +1,231 @@
+
+* `>` - the kv-pair's value is greater than the specified value
+* `>=` - the kv-pair's value is greater than or equal to the specified value
+* `<` - the kv-pair's value is less than the specified value
+* `<=` - the kv-pair's value is less than or equal to the specified value
+
+:::{note}
+There is no `=` operator since `:` functions as an equality operator.
+:::
+
+### Querying array values
+
+To search for a kv-pair where the value is in an array, the syntax is the same as searching for a
+nested kv-pair. For example, the query below...
+
+```
+parent1: {parent2: {child: value}}
+```
+
+...would match the log event below:
+
+```json
+{"parent1": [{"parent2": {"child": "value"}}]}
+```
+
+:::{caution}
+CLP does not currently support queries for array kv-pairs where only part of the key is known. In
+other words, the key must either be a wildcard (`*`) or it must contain no wildcards.
+:::
+
+### Complex queries
+
+You can search for one or more kv-pairs by combining them with boolean algebra. For example:
+
+```text
+key1: value1 AND (key2: valueA OR key2: valueB) AND NOT key3: value3
+```
+
+There are three supported boolean operators:
+
+* `AND` - the expressions on _both_ sides of the operator must be true. 
+* `OR` - the expressions on _either_ side of the operator must be true.
+* `NOT` - the expression after the operator must _not_ be true.
+
+You can use parentheses (`()`) to apply an operator to a group of expressions.
+
+## Examples
+
+**Search for log events that contain a specific key-value pair:**
+
+```
+id: 22149
+```
+
+**Search for ERROR log events containing a substring:**
+
+```
+level: ERROR AND message: "*job*"
+```
+
+**Search for FATAL log events containing the substring "container":**
+
+```
+level: FATAL OR *: *container*
+```
+
+**Search for log events where the value of a nested key is in some range:**
+
+```
+job.stats.latency > 0.5 AND job.stats.latency <= 5
+```
+
+**Search for log events where part of the key is unspecified:**
+
+```
+job.*.status: FAILED
+```
+
+**Search for log events where the value of any child key is "STOPPED":**
+
+```
+job.*: STOPPED
+```
+
+(differences-with-kql)=
+## Differences with KQL
+
+There are a few notable differences between CLP's search syntax and KQL:
+
+* CLP allows a value to contain leading wildcards, by default, whereas they must be explicitly
+  enabled when using KQL with Elasticsearch.
+* CLP doesn't currently support fuzzy matches (e.g., misspellings) for a value, whereas KQL on
+  Elasticsearch may perform a fuzzy match depending on how the kv-pair was ingested.
+* CLP will perform a substring search if the query value contains wildcards or includes spaces,
+  whereas KQL on Elasticsearch may perform a fuzzy match (equivalent to a substring search)
+  depending on how the kv-pair was ingested.
+* CLP doesn't support the following _shorthand_ syntax for matching one or more values with the same
+  key: `key: (value1 or value2)`.
```

Comment:
Hmm this is probably a bug. I think we're supposed to support the key: (OR value1 value2) syntax at least. Probably worth going through and adding unit tests for a lot of the behaviours here within the next few weeks.

Reply:
`key: (OR value1 value2)` works, but where is this syntax from? (I don't see it in the KQL docs). If it's something we defined ourselves, perhaps we can change it to `key: (ANY value1 value2)` to clearly differentiate it from `lhs OR rhs`.

Reply:
Yeah looking through older versions of their docs I'm not seeing it either. Pretty likely I just misinterpreted their grammar when first implementing KQL support.